alphadia/workflow/peptidecentric.py

```diff
@@ -416,77 +416,35 @@ def search_parameter_optimization(self):
                         optimizer.plot()
 
                     break
-                log_string(
-                    f"=== Step {current_step}, extracting elution groups {self.optlock.start_idx} to {self.optlock.stop_idx} ===",
-                    verbosity="progress",
-                )
 
-                feature_df, fragment_df = self.extract_batch(
-                    self.optlock.batch_library.precursor_df,
-                    self.optlock.batch_library.fragment_df,
-                )
-                self.optlock.update_with_extraction(feature_df, fragment_df)
+                log_string(f"Starting optimization step {current_step}.")
```

Comment:
l.402:
my IDE wants me to define `self.optlock` in the constructor, so I guess it also wants you to do the same ;-)

![image](https://github.com/user-attachments/assets/edca87a1-bd82-4d83-95c9-298b162b14cc)

just initialize it with `None` there: it's a good practice to have all instance variables defined at one place

also, check the type annotations: 
![image](https://github.com/user-attachments/assets/adf25d65-f887-4551-8269-d93054e0ae26)

Reply:
now going down that rabbit hole: you might also check the return types of these, the seem off ;-)
![image](https://github.com/user-attachments/assets/6bcfe93b-76d6-48f8-9ff9-ea4aade35756)